index.rst

```diff
@@ -12,290 +12,660 @@ In some cases, this discovery service has already been implemented.
-See :dmtn:`238` for more details.
+There is no one solution that makes sense for every service.
+Options 2 or 3 will make sense for some services but not others, depending on the details of that service, the availability of third-party libraries, and the anticipated use cases of users.
 
-.. _DataLink: https://www.ivoa.net/documents/DataLink/
-.. _SODA: https://www.ivoa.net/documents/SODA/
+In all cases, these helpers should use an underlying service discovery service and library to determine if a given combination of dataset is available in this instance of the Rubin Science Platform and, if so, return one of the four possible objects listed above.
+This hides more complexity from the user, and provides us with more implementation flexibility, than if the user used the discovery service and its library directly.
+For VO services, users could instead query the registry directly with PyVO_, but this is a somewhat complex interface that we want to simplify and all available services may not be registered with IVOA registries or representable in that service discovery system
 
-The second part of the use of DataLink requires sending metadata to the TAP service, so that it knows what DataLink records to return with result rows.
-The DataLink service itself also needs metadata about the TAP schema in order to implement several microservices that provide convenient navigational links from result rows to related TAP queries.
+.. _use-case-sasquatch:
 
-.. _datalink-implementation:
+Sasquatch
+---------
 
-Implementation
---------------
+Sasquatch provides InfluxDB databases for several purposes.
+Writes to those databases by services will generally be done via Kafka, and service discovery for Kafka for services should be done via strimzi-access-operator, thus only requiring service discovery for the Confluent Schema Registry.
+However, query access to the InfluxDB databases from, for example, notebooks will require service discovery of the available databases and their paths and connection information.
 
-Currently, each release of sdm_schemas_ also generates, via GitHub Actions, two additional release artifacts: :file:`datalink-snippets.zip` and :file:`datalink-columns.zip`.
-The TAP service downloads the former on restart and uses it to control which DataLink records are added to result tables.
-The DataLink service downloads the latter on restart and parses it for information about the TAP schema.
+InfluxDB databases require authentication.
+Currently, we use username/password authentication, so a client that wants to query the InfluxDB database needs some mechanism to acquire that password.
+
+Service discovery for Sasquatch InfluxDB databases should then return the following information:
+
+- InfluxDB URL (some clients prefer this as hostname, port, and path, so provide it in both forms)
+- Schema registry URL
```

Comment:
Querying InfluxDB doesn't need the schema registry. The only InfluxDB-related use case that needs the Schema registry is apps serializing messages to put in Kafka that may eventually end up in InfluxDB.

To me, the schema registry conceptually belongs with Kafka, not InfluxDB. Apps putting data into Kafka may need the schema registry no matter where the data eventually ends up. Given the Kafka use case is handled by the Strimzi access operator, maybe the schema registry should be its own service in Repertoire, and not a part of the InfluxDB service.